key.go

```diff
@@ -570,11 +570,21 @@ func readInputs(ctx context.Context, msgs chan<- Msg, input io.Reader) error {
 
 var unknownCSIRe = regexp.MustCompile(`^\x1b\[[\x30-\x3f]*[\x20-\x2f]*[\x40-\x7e]`)
 
+var mouseSGRRegex = regexp.MustCompile(`(\d+);(\d+);(\d+)([Mm])`)
+
 func detectOneMsg(b []byte) (w int, msg Msg) {
 	// Detect mouse events.
+	// X10 mouse events have a length of 6 bytes
 	const mouseEventLen = 6
-	if len(b) >= mouseEventLen && b[0] == '\x1b' && b[1] == '[' && b[2] == 'M' {
-		return mouseEventLen, MouseMsg(parseX10MouseEvent(b))
+	if len(b) >= mouseEventLen && b[0] == '\x1b' && b[1] == '[' {
+		switch b[2] {
+		case 'M':
+			return mouseEventLen, MouseMsg(parseX10MouseEvent(b))
+		case '<':
+			if mouseSGRRegex.Match(b[3:]) {
```

Comment:
You have to fix the mouseEventLen in here otherwise it will throw after each detected sgr event random keyRunes Events with left over chars. Bad for focused input fields and so on. SGR events dont have like X10 Events a fixed length of 6. Otherwise very cool! Works very clean on windows!

My recommended solution:

```go
func detectOneMsg(b []byte) (w int, msg Msg) {
	// Detect mouse events.
	// X10 mouse events have a length of 6 bytes
	const mouseEventX10Len = 6
	if len(b) >= mouseEventX10Len && b[0] == '\x1b' && b[1] == '[' {
		switch b[2] {
		case 'M':
			return mouseEventX10Len, MouseMsg(parseX10MouseEvent(b))
		case '<':
			if matchIndices := mouseSGRRegex.FindSubmatchIndex(b[3:]); matchIndices != nil {
				// SGR mouse events length is the length of the match plus the length of the escape sequence
				mouseEventSGRLen := matchIndices[1] + 3
				return mouseEventSGRLen, MouseMsg(parseSGRMouseEvent(b))
			}
		}
	}
	...
}
 
```

Would love to see this soon be merged into main branch. 😄